src/handlerDecoders.ts

```diff
@@ -0,0 +1,129 @@
+import { decodeAbiParameters } from "viem";
+import { contextType } from "./types";
+import { getToken } from "./utils";
+
+export async function decodeAndSaveStopLoss(
+  staticInput: `0x${string}`,
+  context: contextType,
+  eventId: string
+): Promise<{ stopLossParametersId: string }> {
+  const stopLossData = decodeAbiParameters(
+    [
+      { name: "sellToken", type: "address" },
+      { name: "buyToken", type: "address" },
+      { name: "sellAmount", type: "uint256" },
+      { name: "buyAmount", type: "uint256" },
+      { name: "appData", type: "bytes32" },
+      { name: "receiver", type: "address" },
+      { name: "isSellOrder", type: "bool" },
+      { name: "isPartiallyFillable", type: "bool" },
+      { name: "validityBucketSeconds", type: "uint256" },
+      { name: "sellTokenPriceOracle", type: "bytes32" },
+      { name: "buyTokenPriceOracle", type: "bytes32" },
+      { name: "strike", type: "int256" },
+      { name: "maxTimeSinceLastOracleUpdate", type: "uint256" },
+    ],
+    staticInput
+  );
+  const [tokenIn, tokenOut] = await Promise.all([
+    getToken(stopLossData[0], context),
+    getToken(stopLossData[1], context),
+  ]);
+
+  const StopLossParameters = await context.db.StopLossParameters.create({
+    id: eventId,
+    data: {
+      orderId: eventId,
+      tokenInId: tokenIn.id,
+      tokenOutId: tokenOut.id,
+      tokenAmountIn: stopLossData[2],
+      tokenAmountOut: stopLossData[3],
+      appData: stopLossData[4],
+      to: stopLossData[5],
+      isSellOrder: stopLossData[6],
+      isPartiallyFillable: stopLossData[7],
+      validityBucketSeconds: stopLossData[8],
+      sellTokenPriceOracle: stopLossData[9],
+      buyTokenPriceOracle: stopLossData[10],
+      strike: stopLossData[11],
+      maxTimeSinceLastOracleUpdate: stopLossData[12],
+    },
+  });
+  return { stopLossParametersId: StopLossParameters.id };
+}
+
+export async function decodeAndSaveCoWAmm(
+  staticInput: `0x${string}`,
+  context: contextType,
+  eventId: string
+): Promise<{ cowAmmParametersId: string }> {
+  // on the CoW AMM calls, the first 32 bytes aren't used by us, so we will remove them
```

Comment:
to be very explicit here, you could say we're slicing first 66 chars because it starts with 0x (2 chars) and we want to ignore the following 32 bytes (64 chars)